flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/MailboxProcessor.java

```diff
@@ -457,4 +495,18 @@ private void resumeInternal() {
             }
         }
     }
+
+    /** Isolated API for providing an opportunity to run the mailbox loop which can be suspended. */
+    public final class SuspendableMailboxLoop {
```

Comment:
Maybe we can drop this class? And either move `suspend()` to `MailboxProcessor` or to `org.apache.flink.streaming.runtime.tasks.mailbox.MailboxDefaultAction.Controller`?